ethcore/src/state_db.rs

```diff
@@ -125,14 +172,106 @@ impl StateDB {
  			try!(Self::commit_bloom(batch, bloom_lock.drain_journal()));
  		}
 		let records = try!(self.db.commit(batch, now, id, end));
-		if self.is_canon {
-			self.commit_cache();
-		} else {
-			self.clear_cache();
-		}
+		self.commit_hash = Some(id.clone());
+		self.commit_number = Some(now);
 		Ok(records)
 	}
 
+	/// Apply pending cache changes and synchronize canonical
```

Comment:
maybe "recently committed cache"? if they are on disk then it's completely fair to call it a cache.